### PR TITLE
Use Deployment Environment in Resources' README instead of Environment.

### PR DESCRIPTION
**Pull request by carlosalberto**
Follow up of #606 

CHANGELOG entry is also added.

### CODE REVIEW

**Review by Oberon00 — APPROVED** [review]

**Comment by ecourreges-orange** [comment]
Thanks all for the followup and previous merge, I was on vacation.

**Comment by carlosalberto** [comment]
@bogdandrutu You ok with `Deployment Environment`? ;)

**Review by arminru — APPROVED** [review]

File: specification/resource/semantic_conventions/README.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ Attributes defining a running environment (e.g. Operating System, Cloud, Data Ce
 - [Operating System](./os.md)
 - [Cloud](./cloud.md)
 - Deployment:
-  - [Environment](./deployment_environment.md)
+  - [Deployment Environment](./deployment_environment.md)
```
Comment (bogdandrutu) [inline specification/resource/semantic_conventions/README.md:105]: should then the semantic be:
`deployment_environment.name`?

Comment (Oberon00) [reply specification/resource/semantic_conventions/README.md:105]: I think the semantic convention is probably fine as it is. But maybe the heading structure should actually be:

- Clout
- Deployment
- Kubernetes

without putting Kubernetes under Deployment.

Comment (bogdandrutu) [reply specification/resource/semantic_conventions/README.md:105]: That is also an option :)

**Comment by bogdandrutu** [comment]
Please update the title of the PR

File: specification/resource/semantic_conventions/README.md (hunk 1 above)
Comment (carlosalberto) [reply specification/resource/semantic_conventions/README.md:105]: @Oberon00 Heading structure of this document (README.md) or the convention itself?

Comment (carlosalberto) [reply specification/resource/semantic_conventions/README.md:105]: Ping @Oberon00

Comment (Oberon00) [reply specification/resource/semantic_conventions/README.md:105]: I was talking about the TOC but the actual structure should most likely match that.

Comment (carlosalberto) [reply specification/resource/semantic_conventions/README.md:105]: I'm lost at this moment at what you guys expect ;) Please elaborate on what you want to see here regarding:

1) The TOC.
2) The actual semantic convention.

I don't have a special preference, and personally think the way it stands now (with this PR) is just fine, but I'm open to suggestions.

Comment (Oberon00) [reply specification/resource/semantic_conventions/README.md:105]: I'm also lost 😄

I think the current PR is fine and we can clean this up with #559

**Comment by carlosalberto** [comment]
Merging as this PR has the initial feedback that was left from #606 - remaining bits may be further refined through #559  (as @Oberon00 mentioned).

**Pull request merged**